### PR TITLE
Exercise1

### PR DESCRIPTION
**Pull request by MatzeK105**
Finished exercise 1:
- project setup
- implemented list view
- add and remove comments
- upvote and downvote comments
- comments sorted by score in descending order

Updated READMEs

### CODE REVIEW

**Comment by medizinmensch** [comment]
You got 5 stars in the review below. Furthermore:

* you pulled the new exercises :star: 
* clean responsibilities :star: 
* your PR does not contain unrelated changes :star: 
* could not find any runtime errors :star: 

Thus, you currently have **9 out of 10 stars**.

Where I can not yet give you a point is for **using functional paradigms**. You rather used object an object oriented approach (which can also be fine, but wasn't the task here). You still have a change until next week to change a bit here. 
But maybe @roschaefer is already fine with the implementation like this.

Apart from the points:

What's nice:
What I really appreciate is that you updated your README.md with a how to use and contributors! Very nice!

Room for improvement though:

I prefer having the install/run instructions of the web all at one place only (you have it at `./README.md` and `./weapp/sdf/README.md`). I like it in the `README.md`.

File: exercises/1/README.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
+# VueJS
+
+
+Due date: **28.09.2020**
```
Comment (roschaefer) [inline exercises/1/README.md]: Please update with latest changes: https://github.com/Systems-Development-and-Frameworks/homework/commit/7c99c70634082dcc4579b9622762cb097770ff49

You should "watch" the origin repository as explained in the [README.md](https://github.com/Systems-Development-and-Frameworks/homework#how-to-submit-homework)

watch repo

File: webapp/sdf/src/components/item-component.vue [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+<script>
+import Item from "../classes/item";
+export default {
+  name: "ItemComponent",
+  props: {
+    item: Item,
+  },
+  methods: {
+    removeItem: function (_item) {
```
Comment (roschaefer) [inline webapp/sdf/src/components/item-component.vue]:
```suggestion
    removeItem(_item) {
```

Use shorthand: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Method_definitions#Description

File: webapp/sdf/src/components/item-component.vue [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      this.$emit("removeEvent", _item);
+    },
+
+    updateItem: function (_item, value) {
```
Comment (roschaefer) [inline webapp/sdf/src/components/item-component.vue]:
```suggestion
    updateItem: function (item, value) {
```
Using _underscore prefix for an argument is a naming convention when you don't use the argument in the function body, but you *do use* it on line 23. Other naming conventions use underscores to indicate private methods/properties. But that doesn't seem to be the case here. See: https://eslint.org/docs/rules/no-underscore-dangle

File: webapp/sdf/src/components/list-component.vue [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+  },
+  computed: {
+    sortedArray: function () {
+      var newArray = [...this.itemList];
+
+      return newArray.sort((a, b) => {
+        if (a.votes > b.votes) return -1;
+        if (a.votes < b.votes) return 1;
+        return 0;
+      });
```
Comment (roschaefer) [inline webapp/sdf/src/components/list-component.vue]:
```suggestion
      return newArray.sort((a, b) => b.votes - a.votes);
```

File: webapp/sdf/src/components/list-component.vue [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    };
+  },
+  methods: {
+    removeItem: function (item) {
+      const index = this.itemList.indexOf(item);
+      this.itemList.splice(index, 1);
+    },
+    updateItem: function (item, value) {
+      const index = this.itemList.indexOf(item);
+      this.itemList[index].votes += value;
```
Comment (roschaefer) [inline webapp/sdf/src/components/list-component.vue]: This is not very clean with regards to responsibilities. The list component shouldn't know the internals of an `item`. Instead, you should have a immutable, unique property `id` on the item. Then, list-component can just do:
```js
this.itemList = this.itemList.map((i) => i.id === item.id ? item : i)
```

File: webapp/sdf/src/components/list-component.vue [hunk 6]
```diff
@@ -0,0 +1,75 @@
+<template>
+  <div>
+    <h1>News List</h1>
+
+    <div v-for="(item, index) in sortedArray" :key="index">
+      <ItemComponent
+        v-bind:item="item"
```
Comment (roschaefer) [inline webapp/sdf/src/components/list-component.vue]:
```suggestion
        :item="item"
```
Use shorthand

File: webapp/sdf/src/components/list-component.vue [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+        @updateEvent="updateItem"
+      />
+    </div>
+
+    <div>
+      <form @submit.prevent="onSubmit">
+        <input type="text" v-model="newTitle" />
+        <input type="submit" value="Create" />
+      </form>
+    </div>
```
Comment (roschaefer) [inline webapp/sdf/src/components/list-component.vue]: This could be another component.

File: webapp/sdf/src/components/item-component.vue [hunk 8]
```diff
@@ -0,0 +1,39 @@
+<template>
+  <div>
+    <h2>{{ item.title }} ({{ item.votes }})</h2>
+    <button v-on:click="updateItem(item, 1)">Upvote</button>
+    <button @click="updateItem(item, -1)">Downvote</button>
+    <button v-on:click="removeItem(item)">Remove</button>
```
Comment (roschaefer) [inline webapp/sdf/src/components/item-component.vue]:
```suggestion
    <button @click="removeItem(item)">Remove</button>
```
Use shorthand

**Review by roschaefer — CHANGES_REQUESTED** [review]
nice

Totally agree with what @medizinmensch already reviewed, plus the suggestions below :point_down:

**Comment by medizinmensch** [comment]
I saw you improved by using `filter` instead of splice/index. Good example for functional programming!

You now got 10/10 ⭐ 

Feel free to merge.

**Pull request merged**